crates/e2e/tests/e2e/replace_order.rs

```diff
@@ -270,3 +276,124 @@ async fn single_replace_order_test(web3: Web3) {
         to_wei(3)
     );
 }
+
+async fn single_replace_order_test_with_custom_app_data(web3: Web3) {
```

Comment:
Since this is 99% the same test as `replace_single_order_test` AFAICS and `e2e` tests can already be a bit slow at times I would just adjust `replace_single_order_test` by adding the custom data and asserting that the `fullappdata` of the new order matches your expectation.